README.md

```diff
@@ -1 +1,238 @@
+
+Understanding asynchronous code and its patterns is the key to mastering Node.js. Besides the async code, and a few differences like module system, globals and data types, Node is almost the same as browser JavaScript. For this reason, let's dig deeper into the async code and how we can implement it.
+
+This lesson will cover the sync and async code by using `fs` as an example.
+## Objectives
+
+1. Illustrate sync code with `fs` file read method
+1. Illustrate sync code with `fs` file write method
+1. Illustrate async code wtih fs read method
+1. Illustrate async code wtih fs write method
+
+## Sync Expression
+
+Consider an example in which we have a function and it returns a sum of the values:
+
+```js
+function sum (a, b) {
+  return a+b
+}
+```
+
+This is a synchronous expression, because the function will return the value and the process will be blocked. This means that the control won't be delegated and other tasks cannot be performed.
+
+If the code is simple and fast, as in the case of `a+b`, it's totally fine. Another example would be getting data from a cache. This is fast compare to retrieving the data from a database.
+
+In the synchronous code, we can assign the result of the expression to the variable:
+
+```
+function sum (a, b) {
+  return a+b
+}
+var s = sum(1, 2)
+console.log(s) // 3
+```
+
+The advantages of synchronous code in Node are:
+
+* It's easy to write and read because of familiarity with other programming lanauges
+* It ensures the correct sequence, e.g., `console.log(s)` will alway have the data because it's the last statement and `sum(1, 2)` precedes it.
+* It's appropriate when there are no other tasks running/competing for the same program, i.e., no concurrency, like for database migrations or command-line tool to generate code where you are the only user.
+
+Typically, core Node modules will provide both synchronous and asynchronous methods when applicable so developers can pick the best tool. Let's zoom in on two such methods from `fs`.
+
+## `fs.readFileSync()`
+
+Imagine you are writing a script to extract, transform, and save information from/to a file. The order of the statements matter because you need to extract information first before you can manipulate the data, which needs to happen before you can write to another file. You don't really care about making this script non-blocking because there would be no other requests. It's not a server, it's just a script which has one user (you) and you will launch a maximum of one instances at the same time (concurrently).
+
+To extract the information, we need to read the file `fs.readFileSync`:
+
+```js
+var filename = './fake-path/input.txt'
+var content = fs.readFileSync(filename, options)
+```
+
+If you have console logs, then you can see that the process is blocked until the data `content` becomes available:
```

Comment:
I'm confused by the "if" -- `console` is a global in Node, and `log()` is one of its methods by default.